internal/app/networkd/pkg/address/dhcp.go

```diff
@@ -158,12 +162,14 @@ func (d *DHCP) discover() (*dhcpv4.DHCPv4, error) {
 	// nolint: errcheck
 	defer cli.Close()
 
-	_, ack, err := cli.Request(context.Background(), mods...)
-	if err != nil {
-		// TODO: Make this a well defined error so we can make it not fatal
-		log.Println("failed dhcp request")
-		return nil, err
+	// TODO: this may be better handled by retrying indefinitely in the background
+	for i := 0; i < MaxRequests; i++ {
+		_, ack, err = cli.Request(context.Background(), mods...)
+		if err != nil {
+			// TODO: Make this a well defined error so we can make it not fatal
+			log.Println("failed dhcp request:", err)
+		}
```

Comment:
Maybe we should do something like

`time.Sleep(i * time.Second)`

?

Reply:
The dhcpv4 client internally waits a default of 10s for a timeout... but it's true that in the event of an immediate error, we shouldn't have a tight spin cycle.  I'll put an at-least timer in there.

Reply:
Ahh wasn’t aware that the client already handled it to some degree. SGTM

Reply:
Fixed